index.d.ts

```diff
@@ -44,26 +49,24 @@ declare namespace snakecaseKeys {
   @template Path - Path of keys.
   */
   export type SnakeCaseKeys<
-    T extends Record<string, any> | readonly any[],
+    T extends JsonObjectOptional | readonly any[],
     Deep extends boolean = true,
     Exclude extends readonly unknown[] = EmptyTuple,
     Path extends string = ""
   > = T extends readonly any[]
     ? // Handle arrays or tuples.
       {
-        [P in keyof T]: T[P] extends Record<string, any> | readonly any[]
+        [P in keyof T]: T[P] extends JsonObjectOptional | readonly any[]
         ? SnakeCaseKeys<T[P], Deep, Exclude>
         : T[P];
       }
-    : T extends Record<string, Date | Error | RegExp>
```

Comment:
This condition creates the expected type if, for example

```ts
snakecaseKeys([{ date: new Date() }], { deep: true })
```

but, I think it will not work in the following cases

```ts
snakecaseKeys([{ date: new Date(), fooBar: true }], { deep: true })
```